app/src/main/java/com/example/flickrgallery/data/source/PhotosRoomDataSource.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.flickrgallery.data.source
+
+import com.example.data.source.PhotosLocalDataSource
+import com.example.flickrgallery.db.Db
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import com.example.domain.Photo as DomainPhoto
+import com.example.flickrgallery.model.Photo as RoomPhoto
+
+class PhotosRoomDataSource(database: Db): PhotosLocalDataSource {
+
+    private val photoDao = database.photoDao()
+
+    override suspend fun getAll(): Flow<List<DomainPhoto>> = flow {
```

Comment:
Para no hacer collect, que es una operación terminal, yb volver a emitir propongo algo así:
override fun getAll(): Flow<List<DomainPhoto>> = photoDao.getAllFlow().map {
            it.map { frameworkPhoto -> frameworkPhoto.toDomainPhoto() }
        }

Reply:
OK, entiendo que se puede remapear sin tener que usar algo tan drástico como es hacer un collect.
 Lo cambio en el otro lado también.